eng/common/TestResources/build-test-resource-config.yml

```
@@ -0,0 +1,65 @@
+parameters:
+  - name: SubscriptionConfiguration
+    type: string
+    default: $(sub-config-azure-cloud-test-resources)
+  - name: SubscriptionConfigurations
+    type: object
+    default: null
+
+steps:
+  - ${{ if parameters.SubscriptionConfiguration }}:
+    - pwsh: |
+        $config = @'
+          ${{ parameters.SubscriptionConfiguration }}
+        '@ | ConvertFrom-Json -AsHashtable
+
+        foreach($pair in $config.GetEnumerator()) {
+          if ($pair.Value -is [Hashtable]) {
+            foreach($nestedPair in $pair.Value.GetEnumerator()) {
+              Write-Host "##vso[task.setvariable variable=$($nestedPair.Name);issecret=true;]$($nestedPair.Value)"
+            }
+          } else {
+            Write-Host "##vso[task.setvariable variable=$($pair.Name);issecret=true;]$($pair.Value)"
+          }
+        }
+
+        Write-Host ($config | ConvertTo-Json)
+        $serialized = $config | ConvertTo-Json -Compress
+        Write-Host "##vso[task.setvariable variable=SubscriptionConfiguration;]$serialized"
+      displayName: Initialize SubscriptionConfiguration variable
+
+  - ${{ if parameters.SubscriptionConfigurations }}:
+    - pwsh: |
+        Write-Host "##vso[task.setvariable variable=SubscriptionConfiguration;]{}"
+      displayName: Initialize SubscriptionConfiguration variable
```

```suggestion
      displayName: Initialize SubscriptionConfiguration variable for merging
```

Just something to distinguish it from the earlier step